newsfragments/3081.feature.rst

```diff
@@ -0,0 +1 @@
+Added :func:`trio.lowlevel.add_parking_lot_breaker` and :func:`trio.lowlevel.remove_parking_lot_breaker` to allow creating custom lock/semaphore implementations that will break their underlying parking lot if a task exits unexpectedly. :meth:`trio.lowlevel.ParkingLot.break_lot` is also added, to allow breaking a parking lot intentionally. Breaking a parking lot raises :exc:`trio.BrokenResourceError` for all tasks currently parked in the lot, and any tasks attempting to park in an already broken parking lot will also error. The breakage status of a lot can be viewed and manually modified with the ``trio.ParkingLot.broken_by`` attribute.
```

Comment:
idk why sphinx isn't finding any of the things I added

Reply:
I think they have to be explicitly documented somewhere in a .rst file, since Trio doesn't use `.. automodule::`.

Reply:
ah yes, this sounds familiar. thanks